frame/staking/src/pallet/mod.rs

```diff
@@ -1074,7 +1113,16 @@ pub mod pallet {
 				})
 				.collect::<Result<Vec<_>, _>>()?
 				.try_into()
-				.map_err(|_| Error::<T>::TooManyNominators)?;
+				.defensive_map_err(|_| Error::<T>::TooManyTargets)?;
+
+			ensure!(
+				targets.len() ==
+					targets
+						.iter()
+						.collect::<sp_std::collections::btree_set::BTreeSet<_>>()
```

Comment:
We could do this check while iterating above by building a set. Saves us from iterating the targets a second time to make this set.

Reply:
Although maybe sacrifices readability